streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -219,13 +219,13 @@ public boolean isActive() {
     }
 
     @Override
-    public void maybeRecordRestored(final Time time, final long numRecords) {
-        maybeRecordSensor(numRecords, time, restoreSensor);
-        maybeRecordSensor(-1 * numRecords, time, restoreRemainingSensor);
-    }
-
-    public void initRemainingRecordsToRestore(final Time time, final long numRecords) {
-        maybeRecordSensor(numRecords, time, restoreRemainingSensor);
+    public void recordRestoration(final Time time, final long numRecords, final boolean initRemaining) {
```

Comment:
This is a minor refactor to make sure all sensor recordings are inherited from the `Task` interface so that we do not need to cast to StreamTask.